src/StoreClient.h

```diff
@@ -67,17 +68,23 @@ class store_client
 public:
     store_client(StoreEntry *);
     ~store_client();
-    bool memReaderHasLowerOffset(int64_t) const;
+    /// \returns the smallest of the passed value and the memory offset of the client
+    int memHeaderOffsetLowerThan(const int64_t offset) const;
```

Comment:
For `STORE_MEM_CLIENT` clients, the combination of the method description and implementation effectively equates copyInfo.offset with "the memory offset of the client". However you have [said](https://github.com/measurement-factory/squid/pull/82/files#r383972988) that we cannot define what copyInto.offset is exactly. Can we define copyInfo.offset meaning just for the `STORE_MEM_CLIENT` clients, perhaps?

Or am I reading too much into the method description? Perhaps a386964 simply adjusted this area to hide some data members (the new code appears to be doing what the old code was doing), without really developing the understanding of all these internals? If this is the case, we should probably avoid documenting the new method and disclose our lack of understanding with a TODO comment instead. Since we are not really changing anything in this particular area, it is OK to leave the method mysterious IMO.

Reply:
> However you have said that we cannot define 

No, I meant that 'during one of our discussions it was unclear...'.  And below I clarified it, defining copyInto.offset for HTTP as a "store offset, including range offset, i.e., it is what Http::Stream::getNextRangeOffset() returns". I think we can similarly follow and clarify its value for other paths.

> Perhaps a386964 simply adjusted this area to hide some data members 

Correct, IIRC we decided to hide StoreClient::copyInto, making it private, so that the member(and especially its flags) could not be changed from the outside.

> TODO comment instead

We can do this until we clarify copyInto.offset meaning for all available contexts.